dts/bindings/pinctrl/ambiq,apollo4-pinctrl.yaml

```diff
@@ -108,4 +108,31 @@ child-binding:
         type: int
         default: 0
         description: |
-          IOM nCE module select. The default value is reset value.
+          IOM nCE module select, selects the SPI channel (CE) number (0-3).
+          The default value is 0, which is the reset value of
+          register CFGx.GPIOnOUTCFG. If the pin is not a CE, this
+          descriptor will be ignored.
+      ambiq,iom-mspi:
+        type: int
+        default: 0
+        description: |
+          Indicates the module which uses specific CE pin, 1 if CE is IOM, 0 if MSPI.
+          User should check g_ui8NCEtable in am_hal_gpio.c for the mapping
+          information and config the pins accordingly, we give a default value
+          0 here to make it be consistent with AM_HAL_GPIO_PINCFG_DEFAULT in
+          ambiq hal. If the pin is not a CE, this descriptor will be ignored.
+      ambiq,iom-num:
+        type: int
+        default: 0
+        description: |
+          Indicates the instance which uses specific CE pin.
+          IOM number (0-5) or MSPI (0-2).
+          User should check g_ui8NCEtable in am_hal_gpio.c for the mapping
+          information and config the pins accordingly, we give a default value
+          0 here to make it be consistent with AM_HAL_GPIO_PINCFG_DEFAULT in
```

Comment:
please, do not mention HAL related stuff here. Bindings are not related to HALs.

Reply:
You helped approve these exact changes previously for the apollo3.

https://github.com/zephyrproject-rtos/zephyr/pull/67815/files#diff-86e5617b94d6525f4f7f41747f592977d48d33b9c09e4e20fd2fd35b6fbb9505

Has the Zephyr documentation changed regarding the ability to mention the HAL?
Since it explains HOW to use this feature, I do not understand why it is incorrect to mention the HAL.

Reply:
This was probably an oversight. Users should not have to deal with HAL details when using DT. This is an implementation detail.